progenitor-impl/src/lib.rs

```diff
@@ -26,6 +26,10 @@ pub enum Error {
     TypeError(#[from] typify::Error),
     #[error("unexpected or unhandled format in the OpenAPI document {0}")]
     UnexpectedFormat(String),
+    #[error("invalid file location {0}")]
+    InvalidFileLocation(String),
+    #[error("unrecognised file contents {0}")]
+    InvalidFileContents(String),
```

Comment:
if load_api isn't public we don't need these error cases. I'd prefer copying the code in both progenitor/src/main.rs and progenitor-macro

At some point, progenitor-macro will switch to something that uses include_str!

Reply:
yup, ok :+1: